### PR TITLE
Remove duplicate code

### PR DESCRIPTION
**Pull request by ChenglongGao327**
if user enable debug log, reuse query map in request

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by shankgan — DISMISSED** [review]

**Review by howardjohn — COMMENTED** [review]
I actually don't think we should do this. The risk is that a future refactor will move things around, and subjectToken will now be mutated. This would probably not be detected or fail tests since it only happens at debug enabled, which is not default.

Mutability introduces bugs

**Comment by shankgan** [comment]
> I actually don't think we should do this. The risk is that a future refactor will move things around, and subjectToken will now be mutated. This would probably not be detected or fail tests since it only happens at debug enabled, which is not default.
> 
> Mutability introduces bugs

Want to understand this: As far as I see, there are no changes to behavior introduced by the change. Are you saying that this code could be refactored better? Could that be done in a later PR perhaps? Any future refactor will have to change the code within the debug block with or without this change

**Comment by shankgan** [comment]
> > I actually don't think we should do this. The risk is that a future refactor will move things around, and subjectToken will now be mutated. This would probably not be detected or fail tests since it only happens at debug enabled, which is not default.
> > Mutability introduces bugs
> 
> Want to understand this: As far as I see, there are no changes to behavior introduced by the change. Are you saying that this code could be refactored better? Could that be done in a later PR perhaps? Any future refactor will have to change the code within the debug block, with or without this change

sorry, my bad..I misread this..Agree with your earlier statement

File: security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go [hunk 1]
```diff
@@ -228,22 +228,13 @@ func (p *Plugin) constructFederatedTokenRequest(parameters security.StsRequestPa
...
-		dQuery := map[string]string{
-			"audience":           aud,
-			"grantType":          parameters.GrantType,
-			"requestedTokenType": tokenType,
-			"subjectTokenType":   parameters.SubjectTokenType,
-			"subjectToken":       "redacted",
-			"scope":              reqScope,
-		}
-		dJSONQuery, _ := json.Marshal(dQuery)
+		query["subjectToken"] = "redacted"
```
Comment (shankgan) [inline security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go:231]: nit: would be safer to copy the query map to another map and set the redacted field in that map

**Comment by ChenglongGao327** [comment]
> I actually don't think we should do this. The risk is that a future refactor will move things around, and subjectToken will now be mutated. This would probably not be detected or fail tests since it only happens at debug enabled, which is not default.
> 
> Mutability introduces bugs

yeah，agree  
I just consider that these codes are duplicated

**Pull request closed without merging**